client/src/components/PacmanGame/PacmanGame.jsx

```diff
@@ -43,10 +43,73 @@ class PacmanGame extends Component {
     this.setInitialGameState();
   }
 
+  shouldComponentUpdate(_, newState) {
+    const printDiffs = (objDiffs, debug) => {
+      if (debug && Object.keys(objDiffs).length > 0) {
+        // eslint-disable-next-line no-console
+        console.log('Diff is', (objDiffs));
+      }
+    };
+
+    const onlyPacmanDirectionChange = objDiffs => (
+      Object.keys(objDiffs).length === 1 && objDiffs.pacman && objDiffs.pacman.direction);
+
+    const onlyGhostMoveCountChange = objDiffs => (
+      Object.keys(objDiffs).length === 1 && objDiffs.moveGhostsCount);
```

Comment:
Why is it necessary for Ghosts?

Reply:
We don't need to re-render the canvas if **only the moveGhostCount changes.**  This check prevents the re-render in that case.